google-cloud-clients/google-cloud-storage/src/main/java/com/google/cloud/storage/Bucket.java

```diff
@@ -915,11 +915,18 @@ public Blob create(String blob, byte[] content, String contentType, BlobTargetOp
    * @throws StorageException upon failure
    */
   public Blob create(
-      String blob, InputStream content, String contentType, BlobWriteOption... options) {
+      String blob,
+      Long generation,
+      InputStream content,
+      String contentType,
+      BlobTargetOption... options)
+      throws IOException {
```

Comment:
I don't think you directly change the surface client's methods arguments... this could break users code.
This type of changes are should be discussed. 

@abhinav-qlogic Any particular reason for introducing `Long generation`?